src/sas/qtgui/MainWindow/PackageGatherer.py

```diff
@@ -0,0 +1,278 @@
+                            'zipapp', 'zipfile', 'zipimport', 'zoneinfo', '_aix_support', '_bootlocale',
+                            '_bootsubprocess', '_collections_abc', '_compat_pickle', '_compression', '_markupbase',
+                            '_osx_support', '_pydecimal', '_pyio', '_py_abc', '_sitebuiltins', '_strptime',
+                            '_threading_local', '_weakrefset', '__future__', '__phello__', '__pycache__']
+        standard_lib.extend(sys.builtin_module_names)
+        standard_lib.append("sas")
+
+        for module_name in sys.modules.keys():
+
+            package_name = module_name.split('.')[0]
+
+            # A built in python module or a local file, which have no version, only the python/SasView version
+            if package_name in standard_lib or package_name in package_versions_dict:
+                continue
+
+            # Modules that require specific methods to get the version number
+            if "PyQt5" in module_name:
+                try:
+                    from PyQt5.QtCore import PYQT_VERSION_STR
+                except Exception as e:
+                    # Unable to access PyQt5
+                    err_version_dict[module_name] = f"Unknown: {e} when attempting to get PyQt5 version"
+                    pass
+                else:
+                    package_versions_dict["PyQt"] = PYQT_VERSION_STR
+                try:
+                    from PyQt5.QtCore import QT_VERSION_STR
+                except Exception as e:
+                    # Unable to access Qt
+                    err_version_dict[module_name] = f"Unknown: {e} when attempting to get Qt version"
+                    pass
+                else:
+                    package_versions_dict["Qt"] = QT_VERSION_STR
+                continue
+
+            # Import module
+            try:
+                package = __import__(package_name)
+            except Exception as e:
+                err_version_dict[package_name] = f"Unknown: {e} when attempting to import module"
+                continue
+
+            # Different attempts of retrieving the modules version
+            if hasattr(package, '__version__'):
+                # Module has __version__ attribute
+                try:
+                    package_versions_dict[package_name] = package.__version__
+                except Exception as e:
+                    # Unable to access module
+                    err_version_dict[package_name] = f"Unknown: {e} when attempting to access {package_name} " \
+                                                     f"version using .__version__"
+                    pass
+                continue
+
+            if hasattr(package, '.version'):
+                # Module has .version attribute
+                try:
+                    if isinstance(package.version, str) or isinstance(package.version, float):
+                        package_versions_dict[package_name] = package.version
```

Comment:
If the tests fail then the code falls through to "continue", without listing version for package as "unknown". This will happen for known occurrences of package.version being either a module or a function.

You are presumably coding for conditions that don't exist, otherwise your tests would have resolved these issues before committing. I suggestion removing everything after `package.__version__`, and only add it back in if you know that you need it for an existing sasview dependencies. Since python packages seem to have settled on `__ version__` as the place to store version numbers any new dependencies will probably just work.

You still need a fall-through in case `__version__` isn't found in some future dependency. In those cases just tag the version as unknown. Since this is only for debugging you don't need to go to extraordinary lengths to get versions for every package.

Reply:
Hi @pkienzle 

The: sip, qt5reactor and webencodings packages' version number can only be accessed through pkg_resources, so I think it's worthwhile keeping that code block in, instead of writing specific code for those 3 modules.

I have commented out the .version code block. When I was originally writing the code, there were some modules where .version was required to get their version number. However I cannot remember which modules that was for. Upon testing it seems that not a single SasView imported module triggers this block, so it is redundant. But I can leave the code in just in case as a comment